phoenix-core/src/it/java/org/apache/phoenix/end2end/OnDuplicateKey2IT.java

```diff
@@ -589,10 +577,6 @@ public void testReturnRowResultWithoutAutoCommit() throws Exception {
                     "IGNORE";
             validateReturnedRowAfterUpsert(conn, upsertSql, tableName, 1011.202, null, true,
                     bsonDocument1, bsonDocument1, 123);
-            throw new RuntimeException("Should not reach here");
-        } catch (SQLException e) {
-            Assert.assertEquals(SQLExceptionCode.AUTO_COMMIT_NOT_ENABLED.getErrorCode(),
```

Comment:
Without auto-commit, we won't be able to return ResultSet, correct?

Reply:
Yes, that is why this assumption is no longer valid.